x-pack/plugins/maps/public/classes/styles/vector/components/legend/marker_size_legend.tsx

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component } from 'react';
+import _ from 'lodash';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { EuiFlexGroup, EuiFlexItem, EuiText, EuiToolTip } from '@elastic/eui';
+import { DynamicSizeProperty } from '../../properties/dynamic_size_property';
+
+const FONT_SIZE = 10;
+const HALF_FONT_SIZE = FONT_SIZE / 2;
+const MIN_MARKER_DISTANCE = (FONT_SIZE + 2) / 2;
+
+const EMPTY_VALUE = '';
+
+interface Props {
+  style: DynamicSizeProperty;
+}
+
+interface State {
+  label: string;
+}
+
+export class MarkerSizeLegend extends Component<Props, State> {
+  private _isMounted: boolean = false;
+
+  state: State = {
+    label: EMPTY_VALUE,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._loadLabel();
+  }
+
+  componentDidUpdate() {
+    this._loadLabel();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async _loadLabel() {
+    const field = this.props.style.getField();
+    if (!field) {
+      return;
+    }
+    const label = await field.getLabel();
+    if (this._isMounted && !_.isEqual(this.state.label, label)) {
```

Comment:
nit: I don't think we need to import lodash to compare strings.

```suggestion
    if (this._isMounted && this.state.label !== label {
```